cf/functions.py

```diff
@@ -434,8 +430,22 @@ def _configuration(_Configuration, **kwargs):
         'new_relaxed_identities': relaxed_identities,
         'bounds_combination_mode': bounds_combination_mode,
     }
-    for setting_alias, new_value in kwargs.items():  # for all input kwargs...
-        reset_mapping[setting_alias](new_value)  # ...run corresponding func
+
+    old_values = {}
+
+    try:
+        # Run the corresponding func for all input kwargs
+        for setting_alias, new_value in kwargs.items():
+            reset_mapping[setting_alias](new_value)
+            setting = setting_alias.replace('new_', '', 1)
+            old_values[setting_alias] = old[setting]
+    except ValueError:
```

Comment:
I was wondering if there might be cases missed where exceptions other than `ValueError` are raised, especially here in cf-python with many more constants where some error could be encountered in trying to set them, but it does seem that at the moment it is only possible to run into a `ValueError` (sometimes raised in that form after being caught as e.g. an `AttributeError` or a `TypeError`) so that is all fine.

I'm mainly noting this since, in future, if we add exception handling for the constants they should ultimately raise only `ValueError`, else they should be added to this `except` catch.